docker/daemon/client.go

```diff
@@ -0,0 +1,65 @@
+package daemon
+
+import (
+	"github.com/containers/image/types"
+	"github.com/docker/docker/api"
+	dockerclient "github.com/docker/docker/client"
+	"github.com/docker/go-connections/tlsconfig"
+	"net/http"
+	"path/filepath"
+)
+
+// NewDockerClient initializes a new API client based on the passed SystemContext.
+func newDockerClient(systemCtx *types.SystemContext) (*dockerclient.Client, error) {
+	if systemCtx == nil {
```

Comment:
Non-blocking: I’d prefer the path for `systemCtx == nil` and “`systemCtx.WhateverMember` is unset”, which are defined to be equivalent, to be shared, to minimize duplication and risk of different behaviors.

Reply:
for me this is a guard clause. I prefer to have this out of my way and then being able to write the code knowing that I don't have to deal with _nil_

Reply:
If you want to, guard using `if systemCtx == nil { systemCtx = &types.SystemContext{} }` then.

But I’d really prefer to only have one path with one set of defaults. It already takes non-trivial code inspection to verify that the `httpClient`/`nil` difference is actually equivalent.